#437 Return all rooms when roomName is undefined

lpsandaruwan opened the pull request:
When `roomName ` is not provided, `roomName` has been taken as `matchmake` from the request URI. This PR is fixes #437

endel commented:
Interesting, thanks for the PR @lpsandaruwan! Looking good

- Could you write a small integration test for this? Integration tests are located at `bundles/colyseus/test/IntegrationTest.ts`
- Can you dup this change here as well? 
https://github.com/colyseus/colyseus/blob/master/packages/transport/uwebsockets-transport/src/uWebSocketsTransport.ts#L241-L249

This piece of code is duplicated for `uWebSocketsTransport` - eventually, it would be nice to unify both implementations somehow, as there are other small issues that could benefit from a single implementation (e.g. https://github.com/colyseus/colyseus/issues/439) - the problem with uWebSockets.js is that it does not use the built-in Node.js Server APIs, making it a bit difficult to have a single implementation

Feel free to reach out if you need any information! Cheers

lpsandaruwan commented:
Hi @endel,

Fixed the code as you mentioned and also added the integration test. Please review and let me know if there requires any change.
Thank you! :+1:
